blog/2025-02-03/index.md

```diff
@@ -0,0 +1,67 @@
+---
+slug: flathub-build-infrastructure-revamp
+title: What's next for Flathub build infrastructure
+authors: [barthalion]
+tags: [infrastructure]
+---
+
+There is a storm coming and we are re-architecting our build infrastructure.
+
+<!-- truncate -->
+
+Buildbot has never been designed to do what Flathub needs: taking arbitrary inputs like application IDs and dynamically creating new pipelines. However, there's no misuse one cannot achieve if something is being configured in A Real Programming Language, and so, back in 2019, [Alex Larsson][alex] piled a bunch of hacks so we could have not only dynamic configuration based on Flathub organization in GitHub, but also some custom views displaying latest builds.
+
+Fast-forward to 2025. These hacks no longer work with the latest release of Buildbot, rendering our soft fork stuck on a version from 2021. For whatever reason, updating GitHub CI status stopped to work well, allowing people to merge untested code changes. It also requires periodical restarts because it grinds to a halt for no particular reason, dropping new builds in the meantime. The worst offense: I never liked it.
```

Comment:
```suggestion
Fast-forward to 2025, these hacks no longer work with the latest release of Buildbot, rendering our soft fork stuck on a version from 2021. For whatever reason, updating GitHub CI status stopped to work well, allowing people to merge untested code changes. It also requires periodical restarts because it grinds to a halt for no particular reason, dropping new builds in the meantime. The worst offense: I never liked it.
```